### PR TITLE
sanitizeMap must not compose a geometry collection of a single but non-composable type

### PR DESCRIPTION
**Pull request by BaurzhanSakhariev**
Fixes regression introduced in https://github.com/crate/crate/commit/92127e65a62fcbf2aad67520712f477fa46e63a0

### CODE REVIEW

File: server/src/test/java/io/crate/geo/GeoJSONUtilsTest.java [hunk 1]
```diff
@@ -329,4 +329,23 @@ public void test_geometry_collection_with_nested_collection_single_part_transfor
...
+        Map<String, Object> multiLineString = GeoJSONUtils.wkt2Map(wkt);
+
+        Map<String, Object> collection = Map.of(
+            GeoJSONUtils.TYPE_FIELD,
+            GeoJSONUtils.GEOMETRY_COLLECTION,
+            GeoJSONUtils.GEOMETRIES_FIELD,
+            List.of(multiLineString, multiLineString)
+        );
+        Map<String, Object> sanitizedMap = GeoJSONUtils.sanitizeMap(collection);
+        assertThat(sanitizedMap.get(GeoJSONUtils.TYPE_FIELD)).isEqualTo(GeoJSONUtils.GEOMETRY_COLLECTION);
```
Comment (BaurzhanSakhariev) [inline server/src/test/java/io/crate/geo/GeoJSONUtilsTest.java:345]: Same test passes on master it's a regression. 

We don't have `sanitizeMap` on master, so I used `sanitizeValue` to check.

**Review by seut — CHANGES_REQUESTED** [review]
I do not understand what this code should fix. I also don't see any logical change in the code changes, maybe I'm blind. Please elaborate.

**Comment by BaurzhanSakhariev** [comment]
> I do not understand what this code should fix. I also don't see any logical change in the code changes, maybe I'm blind. Please elaborate.

```
select
  {type='GeometryCollection',
     geometries=[
      {type='MultiLineString', coordinates= [[[100.0, 0.0], [101.0, 1.0] ],[ [102.0, 2.0], [103.0, 3.0]]]},
      {type='MultiLineString', coordinates= [[[100.0, 0.0], [101.0, 1.0] ],[ [102.0, 2.0], [103.0, 3.0]]]}
   ]
  }::GEO_SHAPE
```
              
fails on master with  `ERROR: Cannot cast {"type"='GeometryCollection,...`

I updated my comment above it was confusing (bold stands for the update)

>Same test passes on master **with reverted** https://github.com/crate/crate/commit/92127e65a62fcbf2aad67520712f477fa46e63a0, it's a regression.

> logical change in the code changes

There was no else branch for
```
String targetType = COMPOSABLE_TYPES.get(typeInCollection[0]);
if (targetType != null) {
```
and we returned en empty map.

File: server/src/main/java/io/crate/geo/GeoJSONUtils.java [hunk 2]
```diff
@@ -290,17 +290,15 @@ public static Map<String, Object> sanitizeMap(Map<?, ?> value) {
...
-                    transformed.put(TYPE_FIELD, targetType);
-                    final List<Object> coords = new ArrayList<>();
-                    forEach(geometries, input -> {
-                        Map<String, Object> validatedShape = (Map<String, Object>) input;
-                        coords.add(validatedShape.get(COORDINATES_FIELD));
-                    });
-                    transformed.put(COORDINATES_FIELD, coords);
-                }
+            String targetType;
+            if (sameType[0] == true && (targetType = COMPOSABLE_TYPES.get(typeInCollection[0])) != null) {
```
Comment (seut) [inline server/src/main/java/io/crate/geo/GeoJSONUtils.java:294]: Minor suggestion to improve readability:
```suggestion
            String targetType = COMPOSABLE_TYPES.get(typeInCollection[0]);
            if (sameType[0] == true && targetType != null) {
```

**Comment by seut** [comment]
@BaurzhanSakhariev Thanks for the clarification, did indeed not caught the change in the if/else branch.

**Review by seut — APPROVED** [review]

**Pull request merged**